tardis/io/logger/logger.py

```diff
@@ -1,165 +1,268 @@
 import logging
-import sys
+import os
+import re
+from dataclasses import dataclass, field
+import panel as pn
+from IPython.display import display
+from functools import lru_cache
 
-from tardis.io.logger.colored_logger import ColoredFormatter
+PYTHON_WARNINGS_LOGGER = logging.getLogger("py.warnings")
 
-logging.captureWarnings(True)
-logger = logging.getLogger("tardis")
+@lru_cache(maxsize=1)
+def get_environment():
+    """Determine the execution environment"""
+    try:
+        import IPython
+        ipython = IPython.get_ipython()
 
-console_handler = logging.StreamHandler(sys.stdout)
-console_formatter = ColoredFormatter()
-console_handler.setFormatter(console_formatter)
+        if ipython is None:
+            return 'standard'
 
-logger.addHandler(console_handler)
-logging.getLogger("py.warnings").addHandler(console_handler)
+        # Check for VSCode specific environment variables
+        if any(x for x in ('VSCODE_PID', 'VSCODE') if x in os.environ):
```

Comment:
This may be difficult to maintain. Is this the recommended method?

Reply:
I'm afraid so, I did make some changes to simplifying this in https://github.com/tardis-sn/tardis/pull/2872/commits/669d299bd6032d2ed505338c4380c1a9366b88a9

but detecting vscode was easier than detecting if the code is being run in jupyter notebooks- 
this answer shows how to detect if running in jupyter notebook- https://stackoverflow.com/a/39662359/11974464
and it required accessing internal variables, and this comment here says look for environment variables to check if in vscode- https://github.com/microsoft/vscode-jupyter/issues/7380#issuecomment-912054258 (they are also using panel)